examples/aes-ccm/src/ccm_mode.rs

```diff
@@ -0,0 +1,197 @@
+use std::convert::TryInto;
+use hacspec_lib::*;
+use hacspec_aes::*;
+
+fn format_func(a: &ByteSeq, n: &ByteSeq, p: &ByteSeq, t: u8, alen: u64, nlen: u8, plen: u64) -> ByteSeq {
+    let mut r = 0;
+    let mut tmp = 0;
+
+    if alen < (1<<16 - 1<<8) {
```

Comment:
Why not use the constants (0x800000). The compiler will do that anyway but I also think it's easier to read to have a number there.